components/movementsensor/imuvectornav/imu.go

```diff
@@ -198,38 +198,36 @@ func newVectorNav(
 	if err = v.compensateAccelBias(ctx, pollFreq); err != nil {
 		return nil, err
 	}
-	// compensate for constant DV bias in mesurament
+	// compensate for constant DV bias in measurement
 	if err = v.compensateDVBias(ctx, pollFreq); err != nil {
 		return nil, err
 	}
 
-	if pollFreq > 0 {
-		logger.CDebugf(ctx, "vecnav: will poll at %d Hz", pollFreq)
-		waitCh := make(chan struct{})
-		pollPerSecond := 1.0 / float64(pollFreq)
-		v.workers = goutils.NewBackgroundStoppableWorkers(func(cancelCtx context.Context) {
-			timer := time.NewTicker(time.Duration(pollPerSecond * float64(time.Second)))
-			defer timer.Stop()
-			close(waitCh)
-			for {
-				select {
-				case <-cancelCtx.Done():
-					return
-				default:
-				}
-				select {
-				case <-cancelCtx.Done():
+	logger.CDebugf(ctx, "vecnav: will poll at %d Hz", pollFreq)
```

Comment:
This is fine with me but the previous comment said that we were optionally polling, now we will be polling all the time, is this what we want the behavior to be?

Reply:
I believe so.
Previously, it was "optional polling" on the condition that we had a valid frequency to poll at, i.e. it was not a negative value.
Now that we set by default to have a valid polling frequency it being optional is moot.